vulkan_generator/codegen_utils/codegen.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the utility functions that make generating c++ code nicer"""
+
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import textwrap
+
+def indent(depth : int = 1) -> str: # return the correct indentation for a given indent depth
+ 
```

Comment:
We don't have space between function definition and the implementation in most places. I think they make sense when the function name is very long(there are some in this change for example) but in this case when it's only one line, we shouldn't have them for consistency.

Reply:
I think it's much more readable, but if you have strong feelings I'll remove it.